WalletWasabi/Legal/LegalDocuments.cs

```diff
@@ -10,63 +10,84 @@ namespace WalletWasabi.Legal
 {
 	public class LegalDocuments
 	{
-		public const string EmbeddedFileName = "LegalDocuments.txt";
-		public const string LegalFolderName = "Legal";
-		public const string AssetsFoldername = "Assets";
-		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), LegalFolderName, AssetsFoldername, EmbeddedFileName);
+		public static readonly string EmbeddedFilePath = Path.Combine(EnvironmentHelpers.GetFullBaseDirectory(), "Legal", "Assets", "LegalDocuments.txt");
 
-		public LegalDocuments(string filePath)
+		public LegalDocuments(Version version, string content)
 		{
-			FilePath = Guard.NotNullOrEmptyOrWhitespace(nameof(filePath), filePath, trim: true);
-			var verString = Path.GetFileNameWithoutExtension(FilePath);
-			Version = Version.Parse(verString);
+			Version = version;
+			Content = content;
 		}
 
-		public string FilePath { get; }
 		public Version Version { get; }
+		public string Content { get; }
 
-		public static LegalDocuments TryLoadAgreed(string dataDir)
+		public static async Task<LegalDocuments?> TryLoadAgreedAsync(string folderPath)
```

Comment:
What about just `LoadAgreedDocumentsAsync`?

Reply:
I don't understand too much logic of this function:

* When there is a single file (containing legal docs), you return that document
* When there is no file or when there are multiple files (containing legal docs), you return null. 
  * I would kind of expect that you just return the latest file. That's the source of my confusion.

Reply:
We don't need the older ones. This function ensures there is only one file there - if there is more of something strange that happened and remove all the files, then the client will download it again and fix the issue.